src/Elasticsearch/Framework/AbstractElasticsearchHelper.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Shopware\Elasticsearch\Framework;
+
+use ONGR\ElasticsearchDSL\Search;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityDefinition;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\Plugin\Exception\DecorationPatternException;
+
+abstract class AbstractElasticsearchHelper
+{
+    /**
+     * @throws DecorationPatternException
```

Comment:
I would not mention the throws in the abstract class. as this should do only the core implementation.

Reply:
I rather would. Anyone who decorates it needs to expect that to happen. It is basically part of any decoratable contract.